.github/workflows/Dockerfile

```diff
@@ -1,14 +1,18 @@
-FROM ghcr.io/precice/precice:latest
+ARG PRECICE_TAG=latest
+ARG from=precice/precice:${PRECICE_TAG}
+FROM $from
 
+ARG OPENFOAM_EXECUTABLE=openfoam2112
+ARG OPENFOAM_ADAPTER_REF=develop
 # Install OpenFOAM
-ENV OPENFOAM_VERSION=2112
-RUN wget -q -O - https://dl.openfoam.com/add-debian-repo.sh | sudo bash
-RUN sudo apt-get -qq install openfoam${OPENFOAM_VERSION}-dev
+RUN apt-get update
+RUN wget -q -O - https://dl.openfoam.com/add-debian-repo.sh | bash
+RUN apt-get -qq install $OPENFOAM_EXECUTABLE-dev git
 
 # Bust the cache: we want that the next steps are always executed
 ARG CACHEBUST
 
 # Build the OpenFOAM adapter
-RUN git clone --depth 1 --branch develop https://github.com/precice/openfoam-adapter.git
+RUN git clone --depth 1 --branch $OPENFOAM_ADAPTER_REF https://github.com/precice/openfoam-adapter.git
 WORKDIR /openfoam-adapter
-RUN /usr/bin/openfoam${OPENFOAM_VERSION} ./Allwmake
+RUN /usr/bin/$OPENFOAM_EXECUTABLE ./Allwmake
```

Comment:
I would make the whole path part of the `OPENFOAM_EXECUTABLE`. ESI installs in `/usr/bin`, while Foundation installs in `/opt/`. How how we do it in the custom build workflow: https://github.com/precice/openfoam-adapter/blob/develop/.github/workflows/build-custom.yml

Reply:
Since we currently only plan to support the ESI version i would leave it as be.